.github/workflows/deployment-docker-image.yml

```diff
@@ -50,9 +50,9 @@ jobs:
         uses: ietf-tools/semver-action@v1
         with:
           token: ${{ github.token }}
-          branch: dev
+          branch: ${{ github.ref_name }}
```

Comment:
We always want to use ``dev`` here

Reply:
It's going to get the list of commits between the latest tag and whatever `branch` is set to and using this to determine the version it should use for the build. In my case the latest tag was pointing to the latest commit on `dev`, while the branch I was running the action for had some more commits. This resulted in [the build failing](https://github.com/MaienM/homarr/actions/runs/10985329633/job/30497035910) with the message `Couldn't find any commits between branch HEAD and latest tag.`.

Is this expected? I would expect it to consider the state of the branch selected when triggering the workflow, not `dev`, but maybe I'm misunderstanding how this workflow is intended to be used.

Reply:
I expect this error to happen. We should cancel the workflow run when there is no diff. Do you disagree?
``dev`` is our main development branch. We will never push to master or ``dev`` directly. Hence we should compare on the ``dev`` branch and this workflow should not be executed on different branches.

Reply:
In my case there was a diff though, but it wasn't seeing this diff because it was looking at `dev` instead of the branch it was actually building.

If the intent is that this workflow can only ever work on `dev` this change isn't needed though. It might be worth considering explicitly checking that in a step (as `workflow_dispatch` allows you to select what ref to run for), but that's outside of the scope of this PR.

Reply:
I agree, we should either remove the dispatch or allow runs from other refs. But this PR can be merged as is -all other actions are out of scope. @homarr-labs/maintainers what do you think?